### DISCUSSION TITLE
SP-API Notifications Advice

### DISCUSSION THREAD

**Discussion by belalali0626** [amzn/selling-partner-api-models#4322] [Q&A] [answered]
I'm working on implementing SP-API notifications using SQS and have some questions about seller-specific implementations that aren't clear in the documentation.

**Current Understanding from Docs:**
1. Create a destination (grantless operation)
2. Create subscription using destinationId
3. Set up SQS permissions and KMS if needed

**Key Questions:**
1. How does a subscription get mapped/attached to an individual seller?
   - The createSubscription API doesn't show explicit seller fields
   - No clear indication of how seller authorization is passed

2. For multiple sellers:
   - Do we need separate subscriptions per seller?
   - How does SP-API know which seller a subscription belongs to?
   - How is seller authorization handled in the subscription process?

3. In the SQS queue:
   - How are messages differentiated between sellers?
   - Is seller identification included in the notification payload?

**What I've Tried:**
- Reviewed the notifications documentation
- Set up basic SQS queue and destinations
- Looking at createSubscription API endpoints

Would greatly appreciate any clarification on how the seller-subscription relationship works in practice. Has anyone implemented this for multiple sellers?

Thanks in advance!

**Comment by MeenaAmz** [comment, marked as answer]
[createSubscription](https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-reference#createsubscription) is specific to a selling partner and application. It is NOT a grantless operation and will be using the application credentials and access token used in the call to relate the subscription to the Application-Selling Partner pair. A SubscriptionId is returned as a response to this call. You do need separate subscriptions per seller.

You can differentiate notifications specific to a selling partner in the SQS queue by using SubscriptionId in the NotificationMetadata element of the notification (refer to [Notification structure](https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-use-case-guide#notification-structure))

Thanks
Meena
Selling Partner Developer Services
